include/pybind11/pybind11.h

```diff
@@ -1389,6 +1405,63 @@ class module_ : public object {
         //       For Python 2, reinterpret_borrow was correct.
         return reinterpret_borrow<module_>(m);
     }
+
+    /** \rst
+        Initialized a module def for use with multi-phase module initialization.
+
+        ``def`` should point to a statically allocated module_def.
+        ``slots`` must point to an initialized array of slots with space for at
+           least one additional slot to be populated based on the options.
+    \endrst */
+    template <typename... Options>
+    static object init_module_def(const char *name,
```

Comment:
Maybe use `multi_phase_initialization` for the function name?

IIRC the `_def` was attached for Python 3. It doesn't really mean much anymore.

Reply:
Changed to `initialize_multiphase_module_def` ... the `_def` is slightly important, this doesn't create a PyModule object, it just sets up the def struct, python itself later creates the object (with the `Py_mod_create` slot, which I'm not specifiying here, so we get the default python behavior).